CMakeLists.txt

```diff
@@ -128,6 +107,22 @@ if(LLVM_ENABLE_ZLIB)
   find_package(ZLIB)
 endif()
 
+#-------------------------------------------------------------------------------
+# Performance configuration
+#-------------------------------------------------------------------------------
+
+include(CheckCXXCompilerFlag)
+include(CheckLinkerFlag)
+add_link_options("$<$<BOOL:${STABLEHLO_ENABLE_LLD}>:-fuse-ld=lld>")
+if(STABLEHLO_ENABLE_SPLIT_DWARF)
+    check_cxx_compiler_flag(-gsplit-dwarf STABLEHLO_SUPPORTS_SPLIT_DWARF)
+    add_compile_options("$<$<BOOL:${STABLEHLO_SUPPORTS_SPLIT_DWARF}>:-gsplit-dwarf;-ggnu-pubnames>")
+    check_linker_flag(CXX "-Wl,--gdb-index" STABLEHLO_SUPPORTS_GDB_INDEX)
+    # If we set LLD it doesn't seem to affect the check_linker_flag above.
+    # Account for it with the generator expression OR
+    add_link_options("$<$<OR:$<BOOL:${STABLEHLO_SUPPORTS_GDB_INDEX}>,$<BOOL:${STABLEHLO_ENABLE_LLD}>>:-Wl,--gdb-index>")
```

Comment:
This was kind of confusing -- might open a bug to CMake.
The `check_linker_flag` above doesn't seem to use the previous `add_link_options` to use lld.
The default on my system was using `/bin/ld` (GNU) which didn't support gdb-index.

Reply:
Hmm that is indeed weird. Looks like there's not really a clean way to do this: https://gitlab.kitware.com/cmake/cmake/-/issues/24254

This `OR` LGTM though.

Reply:
https://gitlab.kitware.com/cmake/cmake/-/issues/25649 filed.